worlds/alttp/Options.py

```diff
@@ -110,13 +110,36 @@ class WorldState(Choice):
     option_inverted = 2
 
 
-class Bosses(Choice):
-    option_vanilla = 0
-    option_simple = 1
+class Bosses(TextChoice):
+    """Shuffles bosses around to different locations.
+    Basic will shuffle all bosses except Ganon and Agahnim anywhere they can be placed.
+    Full chooses 3 bosses at random to be placed twice instead of Lanmolas, Moldorm, and Helmasaur.
+    Chaos allows any boss to appear any number of times.
+    Singularity places a single boss in as many places as possible, and a second boss in any remaining locations.
+    Supports plando placement. Formatting here: https://archipelago.gg/tutorial/A%20Link%20to%20the%20Past/plando/en"""
+    display_name = "Boss Shuffle"
+    option_none = 0
+    option_basic = 1
     option_full = 2
     option_chaos = 3
     option_singularity = 4
 
+    def verify(self, world, plando_options):
+        if isinstance(self.value, int):
+            return
+        from Generate import PlandoSettings
+        options = self.value.split(";")
+        if "random" in options:
+            import random
+            options[options.index("random")] = Bosses.name_lookup[random.choice(list(self.name_lookup))]
+            self.value = ";".join(options)
```

Comment:
changing the option in verify seems wrong. this should probably move to the constructor or from_*?